src/ctapipe/monitoring/interpolation.py

```diff
@@ -1,85 +1,81 @@
 from abc import ABCMeta, abstractmethod
+from functools import partial
 from typing import Any
 
 import astropy.units as u
 import numpy as np
 import tables
+from astropy.table import Table
 from astropy.time import Time
 from scipy.interpolate import interp1d
 
 from ctapipe.core import Component, traits
 
 __all__ = [
-    "Interpolator",
+    "MonitoringInterpolator",
+    "LinearInterpolator",
     "PointingInterpolator",
+    "ChunkInterpolator",
 ]
 
 
-class Interpolator(Component, metaclass=ABCMeta):
+class MonitoringInterpolator(Component, metaclass=ABCMeta):
     """
-    Interpolator parent class.
+    MonitoringInterpolator parent class.
 
     Parameters
     ----------
     h5file : None | tables.File
-        A open hdf5 file with read access.
+        An open hdf5 file with read access.
     """
 
-    bounds_error = traits.Bool(
-        default_value=True,
-        help="If true, raises an exception when trying to extrapolate out of the given table",
-    ).tag(config=True)
-
-    extrapolate = traits.Bool(
-        help="If bounds_error is False, this flag will specify whether values outside"
-        "the available values are filled with nan (False) or extrapolated (True).",
-        default_value=False,
-    ).tag(config=True)
-
-    telescope_data_group = None
-    required_columns = set()
-    expected_units = {}
-
-    def __init__(self, h5file=None, **kwargs):
+    def __init__(self, h5file: None | tables.File = None, **kwargs: Any) -> None:
         super().__init__(**kwargs)
 
+        self.telescope_data_group = None
```

Comment:
Why did you change the code here? Why move the class variables to instance variables? 

I don't think the code here should be changed at all. The only thing was to add one "pure" abstract class in the hierarchy. No changes other than changing to that new class as base class should be required for the existing code.

Reply:
At least the interpolators shall be the instance attributes, otherwise they will be shared between the instances and won't be updated if already exist. @ctoennis bumped into this with the tests before. 
Also, @maxnoe do you want to forbid parallel existence of different interpolators?

Reply:
telescope_data_group can be a class variable. the rest needed to be instance variables. I'll make some changes

Reply:
the `telescope_data_group` can be a class variable of some of the child classes, but not of the base class, because in general it is not shared between different interpolators. I.e. if you make it a class variable of the base class, two instances of interpolators could not coexist if the have different `telescope_data_group`.

Reply:
> the rest needed to be instance variables.

Why? It is a class variable now and it works.

Reply:
> At least the interpolators shall be the instance attributes

Yes of course, but they are also instance attributes now